src/App.js

```diff
@@ -139,6 +139,28 @@ const App = () => {
   }
 
   React.useEffect(() => {
+    // Check if the API key is present on the url then put it in the local storage
+    const getAPIKeyFromUrl = () => {
+      const urlParams = new URLSearchParams(window.location.search)
+      const apiKeyParam = urlParams.get('api_key')
+      if (apiKeyParam) {
+        setApiKey(apiKeyParam)
+
+        const testISClient = instantMeilisearch(baseUrl, apiKeyParam, {
+          primaryKey: 'id',
+          clientAgents,
+        })
+        setISClient(testISClient)
+
+        const testMSClient = new Meilisearch({
+          host: baseUrl,
+          apiKey: apiKeyParam,
+          clientAgents,
+        })
+        setMSClient(testMSClient)
```

Comment:
What does this do? Were we before calling these two `setISClient` and `setMSClient` functions, when we were receiving the APIKey from the modal?

Reply:
We where doing it on the [APIKeyModalContent.js](https://github.com/meilisearch/mini-dashboard/blob/main/src/components/ApiKeyModalContent.js#L30) file after the user clicked on the Ok button.  https://github.com/meilisearch/mini-dashboard/blob/6ec828bbf71bf5f51cd006acb9776925d5233a78/src/components/ApiKeyModalContent.js#L30-L46